packages/web-app-files/src/mixins/spaces/actions/setImage.js

```diff
@@ -0,0 +1,84 @@
+import { isLocationSpacesActive } from '../../../router'
+import { clientService } from 'web-pkg/src/services'
+import { mapMutations, mapActions, mapGetters } from 'vuex'
+import { buildResource } from '../../../helpers/resources'
+import { bus } from 'web-pkg/src/instance'
+
+export default {
+  computed: {
+    ...mapGetters(['configuration']),
+    $_setSpaceImage_items() {
+      return [
+        {
+          name: 'set-space-image',
+          icon: 'image-edit',
+          handler: this.$_setSpaceImage_setImageSpace,
+          label: () => {
+            return this.$gettext('Set as space image')
+          },
+          isEnabled: ({ resources }) => {
+            if (resources.length !== 1) {
+              return false
+            }
+            if (!resources[0].mimeType?.startsWith('image/')) {
+              return false
+            }
+            return isLocationSpacesActive(this.$router, 'files-spaces-project')
+          },
+          canBeDefault: false,
+          componentType: 'oc-button',
+          class: 'oc-files-actions-set-space-image-trigger'
+        }
+      ]
+    }
+  },
+  methods: {
+    ...mapMutations('Files', ['UPDATE_RESOURCE_FIELD']),
+    ...mapActions(['showMessage']),
+    async $_setSpaceImage_setImageSpace({ resources }) {
```

Comment:
```suggestion
    async $_setSpaceImage_trigger({ resources }) {
```
to keep it consistent throughout all the action mixins. Makes it easier to refactor later on.